metasmoke.py

```diff
@@ -395,23 +475,22 @@ def stop_autoflagging():
 
     @staticmethod
     def send_statistics():
-        if GlobalVars.metasmoke_down:
+        if GlobalVars.MSStatus.is_down():
             log('warning', "Metasmoke is down, not sending statistics")
             return
 
-        GlobalVars.posts_scan_stats_lock.acquire()
-        if GlobalVars.post_scan_time != 0:
-            posts_per_second = GlobalVars.num_posts_scanned / GlobalVars.post_scan_time
+        (posts_scanned, scan_time, posts_per_second) = GlobalVars.PostScanStat.get_stat()
+        if posts_per_second:
             payload = {'key': GlobalVars.metasmoke_key,
-                       'statistic': {'posts_scanned': GlobalVars.num_posts_scanned, 'api_quota': GlobalVars.apiquota,
+                       'statistic': {'posts_scanned': posts_scanned,
+                                     'api_quota': GlobalVars.apiquota,
                                      'post_scan_rate': posts_per_second}}
         else:
             payload = {'key': GlobalVars.metasmoke_key,
-                       'statistic': {'posts_scanned': GlobalVars.num_posts_scanned, 'api_quota': GlobalVars.apiquota}}
+                       'statistic': {'posts_scanned': posts_scanned,
+                                     'api_quota': GlobalVars.apiquota}}
```

Comment:
Note to self: adjust the above `if`/`else` after it's merged and the merge conflicts are resolved. Probably to:

            payload = {'key': GlobalVars.metasmoke_key,
                       'statistic': {'posts_scanned': GlobalVars.num_posts_scanned, 'api_quota': current_apiquota,
                                     'post_scan_rate': posts_per_second}}
            if not posts_per_second:
                del payload['statistic']['post_scan_rate']